File: controller/nests/delete_item.php
<?php
namespace Core;
use Controller\files\UtilForFiles;
use Exception;

if (!defined('__GOOSE__')) exit();

/**
 * delete nest
 *
 * @var Goose $this
 */

try
{
  // check and set srl
  $srl = (int)$this->params['srl'];
  if (!($srl && $srl > 0))
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Controller::checkAccessItem((object)[
    'model' => $this->model,
    'table' => 'nests',
    'srl' => $srl,
  ]);

  // remove articles
  $articles = $this->model->getItems((object)[
    'table' => 'articles',
    'field' => 'srl',
    'where' => 'nest_srl='.$srl,
  ]);
  if ($articles->data && count($articles->data))
  {
    foreach($articles->data as $k=>$v)
    {
      // remove thumbnail image
      UtilForFiles::removeThumbnailImage($this->model, $v->srl);
      // remove files
      UtilForFiles::removeAttachFiles($this->model, $v->srl);
    }
    // remove articles
    $this->model->delete((object)[
      'table' => 'articles',
      'where' => 'nest_srl='.$srl,
    ]);
  }

  // remove categories
  $categoriesCount = $this->model->getCount((object)[
    'table' => 'categories',
    'where' => 'nest_srl='.$srl,
  ]);
  if ($categoriesCount->data > 0)
  {
    $this->model->delete((object)[
      'table' => 'categories',
      'where' => 'nest_srl='.$srl,
    ]);
  }

  // remove nest
  $output = Controller::delete((object)[
    'model' => $this->model,
    'table' => 'nests',
    'srl' => $srl,
  ]);

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  Output::data($output);
}
catch (Exception $e)
{
  $this->model->disconnect();
  Error::data($e->getMessage(), $e->getCode());
}

File: src/endpoints/tag/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from . import __libs__ as tag_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _check_token = True):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        if _check_token: checking_token(req, db)

        # check module data
        count = db.get_count(
            table_name=tag_libs.Module.get_table_name(params.module),
            where=[ f'srl = {params.module_srl}' ],
        )
        if not (count > 0): raise Exception('Module data not found.', 204)

        # delete data
        tag_libs.delete(db, params.module, params.module_srl)

        # set result
        result = output.success({
            'message': 'Complete delete tag.',
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result
